rpc_cli/src/lib.rs

```diff
@@ -0,0 +1,167 @@
+				for r in reqs {
+					sign_interactive(signer, &password, r)
+				}
+				Ok("".to_owned())
+			}
+			Err(err) => {
+				Err(format!("error: {:?}", err))
+			}
+		}
+	}).map_err(|err| {
+		format!("{:?}", err)
+	}).wait())
+}
+
+fn list_transactions(signer: &mut SignerRpc) -> Result<String, String> {
+	try!(signer.requests_to_confirm().map(|reqs| {
+		match reqs {
+			Ok(ref reqs) if reqs.is_empty() => {
+				Ok("No transactions in signing queue".to_owned())
+			}
+			Ok(ref reqs) => {
+				Ok(format!("Transaction queue:\n{}", reqs
+						   .iter()
+						   .map(|r| format!("{}", r))
+						   .collect::<Vec<String>>()
+						   .join("\n")))
+			}
+			Err(err) => {
+				Err(format!("error: {:?}", err))
+			}
+		}
+	}).map_err(|err| {
+		format!("{:?}", err)
+	}).wait())
+}
+
+fn sign_transaction(signer: &mut SignerRpc,
+					id: U256,
+					password: &str) -> Result<String, String> {
+	try!(signer.confirm_request(id, None, password).map(|res| {
+		match res {
+			Ok(u) => Ok(format!("Signed transaction id: {:#x}", u)),
+			Err(e) => Err(format!("{:?}", e)),
+		}
+	}).map_err(|err| {
+		format!("{:?}", err)
+	}).wait())
+}
+
+fn reject_transaction(signer: &mut SignerRpc,
+					  id: U256) -> Result<String, String> {
+	try!(signer.reject_request(id).map(|res| {
+		match res {
+			Ok(true) => Ok(format!("Rejected transaction id {:#x}", id)),
+			Ok(false) => Err(format!("No such request")),
+			Err(e) => Err(format!("{:?}", e)),
+		}
+	}).map_err(|err| {
+		format!("{:?}", err)
+	}).wait())
+}
+
+// cmds
+
+pub fn cmd_signer_list(signerport: u16,
+					   authfile: PathBuf) -> Result<String, String> {
+	let mut signer = try!(SignerRpc::new(&format!("ws://127.0.0.1:{}", signerport), &authfile).map_err(|err| {
+		format!("{:?}", err)
+	}));
+	list_transactions(&mut signer)
+}
+
+pub fn cmd_signer_reject(id: Option<usize>, signerport: u16,
+						 authfile: PathBuf) -> Result<String, String> {
+	let id = try!(id.ok_or(format!("id required for signer reject")));
+	let mut signer = try!(SignerRpc::new(&format!("ws://127.0.0.1:{}", signerport), &authfile).map_err(|err| {
+		format!("{:?}", err)
+	}));
+	reject_transaction(&mut signer, U256::from(id))
+}
+
+pub fn cmd_signer_sign(id: Option<usize>,
+					   pwfile: Option<PathBuf>,
+					   signerport: u16,
+					   authfile: PathBuf) -> Result<String, String> {
+	let password;
+	match pwfile {
+		Some(pwfile) => {
+			match File::open(pwfile) {
+				Ok(fd) => {
+					match BufReader::new(fd).lines().next() {
+						Some(Ok(line)) => password = line,
+						_ => return Err(format!("No password in file"))
+					}
+				},
+				Err(e) => return Err(format!("Could not open password file: {}", e))
+			}
+		}
+		None => {
+			password = match rpassword::prompt_password_stdout("Password: ") {
+				Ok(p) => p,
+				Err(e) => return Err(format!("{}", e)),
+			}
+		}
+	}
+
+	let mut signer = try!(SignerRpc::new(&format!("ws://127.0.0.1:{}", signerport), &authfile).map_err(|err| {
+		format!("{:?}", err)
```

Comment:
There is dozens of `format!("{:?}", err)` in that file. I think it would be worth to do:

```rust
struct SignError(String);
impl<T: fmt::Debug> From<T> for SignError {
  fn from(err: T) -> Self {
    format!("{:?}", err)
  }
} 
```

You could simply `try!` in most of the places then.